app/controllers/foreman_remote_execution/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -0,0 +1,31 @@
+module ForemanRemoteExecution
+  module Concerns
+    module Api::V2::RegistrationControllerExtensions
+      module ApipieExtensions
+        extend Apipie::DSL::Concern
+
+        update_api(:global, :host) do
+          param :remote_execution_interface, String, desc: N_("Identifier of the Host interface for Remote execution")
+        end
+      end
+
+      extend ActiveSupport::Concern
+
+      def host_setup_extension
+        remote_execution_interface
+        super
+      end
+
+      def remote_execution_interface
+        return unless params['remote_execution_interface'].present?
+
+        interfaces = @host.interfaces
+        interfaces.find_by!(identifier: params['remote_execution_interface'])
```

Comment:
this results in ugly error message if the interface with such identifier does exist or is not created for any reason (e.g. sub-man does not send facts), proably could be a problem for debian/ubuntu. We should try to find the interface and fail if more descriptive exception message if the interface is nil.